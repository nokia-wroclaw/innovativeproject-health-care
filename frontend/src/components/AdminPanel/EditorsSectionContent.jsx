import React from "react";
import { connect } from "react-redux";
import { deleteEditor } from "./../../store/actions/editors";
import { Card, Grid } from "semantic-ui-react";
import colors from "../../styles/colors";
import SectionIcon from "../common/SectionIcon/";

const EditorsSectionContent = props => {
  return (
    <Card.Content>
      <Grid>
        <Grid.Row columns={2}>
          <Grid.Column width={13} floated="left">
            {props.editor.name}
          </Grid.Column>
          <Grid.Column width={3} floated="right" textAlign="center">
            <SectionIcon
              className="fa fa-minus-square-o"
              color={colors.red}
              onClick={() => props.deleteEditor(props.editor)}
            />
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Card.Content>
  );
};

export default connect(
  null,
  { deleteEditor }
)(EditorsSectionContent);
