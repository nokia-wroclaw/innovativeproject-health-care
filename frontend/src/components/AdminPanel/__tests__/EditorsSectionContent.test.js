import React from "react";
import { shallow, mount } from "enzyme";
import { EditorsSectionContent } from "../EditorsSectionContent";

describe("EditorsSectionContent", () => {
  it("should match snapshot", () => {
    expect(
      shallow(<EditorsSectionContent editor={{ name: "John" }} />)
    ).toMatchSnapshot();
  });

  it("should have editor prop and it should be an object with 'name' field", () => {
    const wrapper = mount(<EditorsSectionContent editor={{ name: "John" }} />);
    expect(wrapper.prop("editor")).toBeInstanceOf(Object);
    expect(wrapper.prop("editor").name).toBeTruthy();
  });
});
