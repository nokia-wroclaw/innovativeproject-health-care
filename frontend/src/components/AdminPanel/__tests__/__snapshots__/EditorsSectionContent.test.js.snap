// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorsSectionContent should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditorsSectionContent
    editor={
      Object {
        "name": "John",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Grid>
        <GridRow
          columns={2}
        >
          <GridColumn
            floated="left"
            width={13}
          >
            John
          </GridColumn>
          <GridColumn
            floated="right"
            textAlign="center"
            width={3}
          >
            <SectionIcon
              className="fa fa-minus-square-o"
              color="#E80000"
              onClick={[Function]}
            />
          </GridColumn>
        </GridRow>
      </Grid>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <GridRow
          columns={2}
        >
          <GridColumn
            floated="left"
            width={13}
          >
            John
          </GridColumn>
          <GridColumn
            floated="right"
            textAlign="center"
            width={3}
          >
            <SectionIcon
              className="fa fa-minus-square-o"
              color="#E80000"
              onClick={[Function]}
            />
          </GridColumn>
        </GridRow>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GridColumn
              floated="left"
              width={13}
            >
              John
            </GridColumn>,
            <GridColumn
              floated="right"
              textAlign="center"
              width={3}
            >
              <SectionIcon
                className="fa fa-minus-square-o"
                color="#E80000"
                onClick={[Function]}
              />
            </GridColumn>,
          ],
          "columns": 2,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "John",
              "floated": "left",
              "width": 13,
            },
            "ref": null,
            "rendered": "John",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <SectionIcon
                className="fa fa-minus-square-o"
                color="#E80000"
                onClick={[Function]}
              />,
              "floated": "right",
              "textAlign": "center",
              "width": 3,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "fa fa-minus-square-o",
                "color": "#E80000",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Grid>
          <GridRow
            columns={2}
          >
            <GridColumn
              floated="left"
              width={13}
            >
              John
            </GridColumn>
            <GridColumn
              floated="right"
              textAlign="center"
              width={3}
            >
              <SectionIcon
                className="fa fa-minus-square-o"
                color="#E80000"
                onClick={[Function]}
              />
            </GridColumn>
          </GridRow>
        </Grid>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <GridRow
            columns={2}
          >
            <GridColumn
              floated="left"
              width={13}
            >
              John
            </GridColumn>
            <GridColumn
              floated="right"
              textAlign="center"
              width={3}
            >
              <SectionIcon
                className="fa fa-minus-square-o"
                color="#E80000"
                onClick={[Function]}
              />
            </GridColumn>
          </GridRow>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <GridColumn
                floated="left"
                width={13}
              >
                John
              </GridColumn>,
              <GridColumn
                floated="right"
                textAlign="center"
                width={3}
              >
                <SectionIcon
                  className="fa fa-minus-square-o"
                  color="#E80000"
                  onClick={[Function]}
                />
              </GridColumn>,
            ],
            "columns": 2,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "John",
                "floated": "left",
                "width": 13,
              },
              "ref": null,
              "rendered": "John",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <SectionIcon
                  className="fa fa-minus-square-o"
                  color="#E80000"
                  onClick={[Function]}
                />,
                "floated": "right",
                "textAlign": "center",
                "width": 3,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "fa fa-minus-square-o",
                  "color": "#E80000",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
